cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/ClusterBootstrap.scala

```diff
@@ -33,13 +37,28 @@ final class ClusterBootstrap(implicit system: ExtendedActorSystem) extends Exten
   val settings = ClusterBootstrapSettings(system.settings.config)
 
   // used for initial discovery of contact points
-  val discovery: ServiceDiscovery = {
-    val clazz = settings.contactPointDiscovery.discoveryClass
-    system.dynamicAccess.createInstanceFor[ServiceDiscovery](clazz, List(classOf[ActorSystem] → system)).get
-  }
+  val discovery: SimpleServiceDiscovery =
+    settings.contactPointDiscovery.discoveryMethod match {
+      case "akka.discovery" ⇒
+        val discovery = ServiceDiscovery(system).discovery
+        log.info("Bootstrap using default `akka.discovery` mechanism: {}", Logging.simpleName(discovery))
+        discovery
+
+      case otherDiscoveryMechanism ⇒
+        val implClazz = system.settings.config.getString(otherDiscoveryMechanism + ".impl")
+        log.info("Bootstrap using [{}] discovery mechanism, instantiating [{}]", otherDiscoveryMechanism, implClazz)
+        system.dynamicAccess
+          .createInstanceFor[SimpleServiceDiscovery](implClazz, List(classOf[ActorSystem] → system))
+          .get
+    }
```

Comment:
this allows for two things.
since discovery is "just a tool", but we're using it here, if someone does not configure anything in bootstrap, but they have picked some discovery impl, THAT discovery impl will be used by default here.
if they want to use DNS here, but something else for "discovery" everywhere else they can configure that too

Reply:
People can use `ServiceDiscovery().locate("thingy")` just in apps to locate anything they want.
So maybe they do want to do bootstrap using DNS, but other discovery they want ZK or whatever, this is to enable that usecase